internal/service/connect/hours_of_operation.go

```diff
@@ -0,0 +1,352 @@
+		input.TimeZone = aws.String(d.Get("time_zone").(string))
+	}
+
+	_, err = conn.UpdateHoursOfOperationWithContext(ctx, input)
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("[ERROR] Error updating HoursOfOperation (%s): %w", d.Id(), err))
+	}
+
+	if d.HasChange("tags_all") {
+		o, n := d.GetChange("tags_all")
+		if err := UpdateTags(conn, d.Id(), o, n); err != nil {
+			return diag.FromErr(fmt.Errorf("error updating tags: %w", err))
+		}
+	}
+
+	return resourceHoursOfOperationRead(ctx, d, meta)
+}
+
+func resourceHoursOfOperationDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).ConnectConn
+
+	instanceID, hoursOfOperationID, err := HoursOfOperationParseID(d.Id())
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, err = conn.DeleteHoursOfOperationWithContext(ctx, &connect.DeleteHoursOfOperationInput{
+		HoursOfOperationId: aws.String(hoursOfOperationID),
+		InstanceId:         aws.String(instanceID),
+	})
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error deleting HoursOfOperation (%s): %w", d.Id(), err))
+	}
+
+	return nil
+}
+
+func expandConfigs(configs []interface{}) []*connect.HoursOfOperationConfig {
+	if len(configs) == 0 {
+		return nil
+	}
+
+	hoursOfOperationConfigs := []*connect.HoursOfOperationConfig{}
+	for _, config := range configs {
+		data := config.(map[string]interface{})
+		hoursOfOperationConfig := &connect.HoursOfOperationConfig{
+			Day: aws.String(data["day"].(string)),
+		}
+
+		tet := data["end_time"].([]interface{})
+		vet := tet[0].(map[string]interface{})
+		et := connect.HoursOfOperationTimeSlice{
+			Hours:   aws.Int64(int64(vet["hours"].(int))),
+			Minutes: aws.Int64(int64(vet["minutes"].(int))),
+		}
+		hoursOfOperationConfig.EndTime = &et
+
+		tst := data["start_time"].([]interface{})
+		vst := tst[0].(map[string]interface{})
+		st := connect.HoursOfOperationTimeSlice{
+			Hours:   aws.Int64(int64(vst["hours"].(int))),
+			Minutes: aws.Int64(int64(vst["minutes"].(int))),
+		}
+		hoursOfOperationConfig.StartTime = &st
+
+		hoursOfOperationConfigs = append(hoursOfOperationConfigs, hoursOfOperationConfig)
+	}
+
+	return hoursOfOperationConfigs
+}
+
+func flattenConfigs(configs []*connect.HoursOfOperationConfig) []interface{} {
+	configsList := []interface{}{}
+	for _, config := range configs {
+		values := map[string]interface{}{}
+		values["day"] = aws.StringValue(config.Day)
+
+		et := map[string]interface{}{
+			"hours":   aws.Int64Value(config.EndTime.Hours),
+			"minutes": aws.Int64Value(config.EndTime.Minutes),
+		}
+		values["end_time"] = []interface{}{et}
+
+		st := map[string]interface{}{
+			"hours":   aws.Int64Value(config.StartTime.Hours),
+			"minutes": aws.Int64Value(config.StartTime.Minutes),
+		}
+		values["start_time"] = []interface{}{st}
+		configsList = append(configsList, values)
+	}
+	return configsList
+}
+
+func HoursOfOperationParseID(id string) (string, string, error) {
```

Comment:
can you please rename this to `DecodeHoursOfOperationId`

Reply:
After discussions, for consistency with other connect resources, we will retain the current naming. 

https://github.com/hashicorp/terraform-provider-aws/blob/b5f182feced587e2715478c045ef5f78eba7f84d/internal/service/connect/contact_flow.go#L287